hivemind/p2p/p2p_daemon.py

```diff
@@ -437,20 +419,10 @@ def iterate_protobuf_handler(
     def _start_listening(self) -> None:
         async def listen() -> None:
             async with self._client.listen():
-                await self._server_stopped.wait()
+                await asyncio.Future()  # Wait until this task will be cancelled in _terminate()
 
         self._listen_task = asyncio.create_task(listen())
 
-    async def _stop_listening(self) -> None:
```

Comment:
Cancelling `self._listen_task` is enough here. The corresponding code moved to `self._terminate()` (since it is not async anymore).